tests/kernel/threads/dynamic_thread_stack/src/main.c

```diff
@@ -111,6 +112,8 @@ ZTEST(dynamic_thread_stack, test_dynamic_thread_stack_alloc)
 		}
 	}
 
+	sys_cache_data_flush_all();
```

Comment:
Should add a comment explaining the specific coherence problems on intel_adsp, otherwise this looks like voodoo

Reply:
that is fair. Btw, just checked that this is not enough if the thread is allocating these stacks switches to a different cpu before flush the cache.